code/__HELPERS/unsorted.dm

```diff
@@ -1257,3 +1257,23 @@ will handle it, but:
 	return	cone_turfs
 
 GLOBAL_LIST_INIT(survivor_outfits, typecacheof(/datum/outfit/job/survivor))
+
+/**
+ *	Draws a line between two atoms, then checks if the path is blocked.
+ *	Variables:
+ *	start -start point of the path
+ *	end - end point of the path
+ *	bypass_window - whether it will go through transparent windows in the same way as lasers
+ *	projectile - whether throwpass will be checked to ignore dense objects in the same way as projectiles
+ *	bypass_xeno - whether to bypass dense xeno structures in the same way as flamers
+ */
+/proc/check_path(atom/start, atom/end, bypass_window = FALSE, projectile = FALSE, bypass_xeno = FALSE)
+	var/list/path_to_target = getline(start, end)
+	var/line_count = 1
+	for(var/path_turf in path_to_target)
+		if(line_count == length(path_to_target))
+			break
+		if(LinkBlocked(path_to_target[line_count], path_to_target[line_count + 1], bypass_window, projectile, bypass_xeno))
+			return FALSE
+		line_count ++
```

Comment:
why does line_count exist if this isnt a while loop?

Reply:
made it a while since that basically what I was going for, unless you think it's better to do otherwise.

Reply:
I'll fix tomorrow.